### PR TITLE
[IMP] inventory: invoice shipping

### PR DESCRIPTION
**Pull request by Felicious**
Update 8 year old doc about invoicing shipping costs

FWport: yes (:

[Task](https://www.odoo.com/web#id=3576562&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form)

### CODE REVIEW

File: content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst [hunk 1]
```diff
@@ -1,87 +1,106 @@
...
-.. image:: invoicing/invoicing03.png
-   :align: center
+Invoice real shipping costs
+===========================
 
-When you create the invoice, it will take the price computed by the
-carrier.
+To modify the invoice to reflect the real cost of shipping, follow the steps :ref:`above
+<inventory/shipping/invoice-so>` to create an invoice with a delivery product with a :guilabel:`Unit
+Price` of zero.
```
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst:96]: The steps above are about creating an SO, not an invoice, which seems to be what we'd want to do here first.
```suggestion
<inventory/shipping/invoice-so>` to create and and confirm a sales order with a delivery product with a :guilabel:`Unit
Price` of zero.
```

File: content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst [hunk 2]
```diff
@@ -1,87 +1,106 @@
...
 
-Go back to the sale order, the real cost is now added to the sale
-order.
+Finally, after the product is delivered, click :guilabel:`Create invoice`, and an invoice is created
+that includes the shipping cost that was added earlier.
 
-.. image:: invoicing/invoicing03.png
-   :align: center
+Invoice real shipping costs
+===========================
```
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst:92]: Overall suggestion for this section is to add more UI walk-through. We *tell* more than *show* here and I'm confident this feedback would come up in KC's review anw 😉

**Review by StraubCreative — CHANGES_REQUESTED** [review]
Hi @Felicious 
Great job catching this old-timey doc and performing such a clear and easy-to-follow rewrite. I just have two necessary changes below, and the rest are optional, if you could take a look pleace. Can tag me again for merge, thanks!

File: content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst [hunk 3]
```diff
@@ -1,87 +1,106 @@
...
+Next, configure the the price of each delivery method by going to :menuselection:`Inventory app -->
+Configuration --> Shipping Methods` and click the :guilabel:`Create` button. Doing so opens a form
+to provide details about the shipping provider, including:
 
-Invoice the price set on the sale order
----------------------------------------
+- :guilabel:`Shipping Method` (*Required field*): the name of the delivery method (e.g. `flat-rate
+  shipping`, `same day delivery`, etc.).
+- :guilabel:`Provider` (*Required field*): choose the delivery service, like Fedex, if using a
+  :doc:` third-party carrier <../setup/third_party_shipper>` Ensure the integration with the
```
[the remark this answers is not in the document]
Comment (Felicious) [reply content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst:37]: i added an extra space! fixed

File: content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst [hunk 4]
```diff
@@ -1,87 +1,106 @@
...
+  shipping carrier is properly installed and select the provider from the drop-down menu.
```
[the remark this answers is not in the document]
Comment (Felicious) [reply content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst:38]: added an admonition block to link to the third party integration

File: content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst [hunk 5]
```diff
@@ -1,87 +1,106 @@
...
-On your sale order, choose the carrier that will be used. Click on
-**Delivery Method** to choose the right one.
+Then, click the :guilabel:`Get Rate` button to the calculate shipping price based on real-time
+shipping data Odoo's shipping carrier integration.
 
-.. image:: invoicing/invoicing05.png
+The :guilabel:`Cost` is automatically calculated using the weight of the items in the order.
+Finally, click the :guilabel:`Add` button to close the window.
+
+.. image:: invoicing/add-a-shipping-method.png
```
[the remark this answers is not in the document]
Comment (Felicious) [reply content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst:74]: updated the image to address this feedback (:

File: content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst [hunk 6]
```diff
@@ -1,87 +1,106 @@
...
+.. _inventory/shipping/sales-order:
 
-The real shipping cost is computed when the delivery order is
-validated, you can see the real cost in the chatter of the delivery order.
+On the sales order, the delivery product appears in the :guilabel:`Order Lines` tab, with the
+:guilabel:`Unit Price` set as the shipping cost calculated in the :guilabel:`Add a shipping method`
+pop-up window.
 
-.. image:: invoicing/invoicing02.png
+.. image:: invoicing/delivery-product.png
```
[the remark this answers is not in the document]
Comment (Felicious) [reply content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst:84]: updated image to address feedback (:

File: content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst (hunk 1 above)
Comment (Felicious) [reply content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst:96]: I'm not sure if I agree? The steps above take a longer time adding the delivery product on the SO, yes, but the last instruction is to create an invoice. In line 88:
> Finally, after the product is delivered, click :guilabel:`Create invoice`, and an invoice is created
that includes the shipping cost that was added earlier.

**Comment by Felicious** [comment]
Hi @StraubCreative ! Thank you for the review and pointing everything out!

I addressed most of them, and I'll leave the rest for when I come back, in a follow-up PR.

**Review by StraubCreative — APPROVED** [review]
Thanks for the quick revisions @Felicious 
Indeed, we can make further revisions in another PR— for now let's ship the updates on this old doc.

cc: @ksc-odoo @samueljlieber I fast-tracked this PR since it was so outdated. Counting as a new contribution.
.....
@robodoo r+

**Comment by StraubCreative** [comment]
@robodoo retry

**Pull request closed without merging**